dist/performance.js

```diff
@@ -11,34 +23,105 @@ function getLcpElement() {
             resolve(naiveLcpEntry);
         }).observe({ type: "largest-contentful-paint", buffered: true });
     }).then(({ startTime, element, url, size, loadTime, renderTime }) => {
-        let attributes = [];
-        for (let index = 0; index < element.attributes.length; index++) {
-            const ele = element.attributes.item(index);
-            attributes[index] = { name: ele.name, value: ele.value };
-        }
-
+        const attributes = getAttributes(element);
+        const styles = getComputedStyles(element, ['background-image']);
         return {
             startTime,
-            nodeName: element.nodeName,
+            nodeName: element?.nodeName,
             url,
             size,
             loadTime,
             renderTime,
             attributes,
+            boundingClientRect: element?.getBoundingClientRect().toJSON(),
+            naturalWidth: element?.naturalWidth,
+            naturalHeight: element?.naturalHeight,
+            styles
+        };
+    });
+}
+
+function getAttributes(element) {
+    if (!element) {
+        return null;
+    }
+
+    return Array.from(element.attributes).map(attr => {
+        return {
+            name: attr.name,
+            value: attr.value
         };
     });
 }
 
+function getComputedStyles(element, properties) {
+    if (!element) {
+        return null;
+    }
+
+    const styles = getComputedStyle(element);
+    return Object.fromEntries(properties.map(prop => ([prop, styles.getPropertyValue(prop)])));
+}
+
+function summarizeLcpElement(element) {
+    if (!element) {
+        return null;
+    }
+
+    const nodeName = element.nodeName;
+    const attributes = getAttributes(element);
+
+    return {
+        nodeName,
+        attributes
+    }
+}
+
 function getWebVitalsJS() {
     const webVitalsJSPattern = /(8999999999999[\s\S]+1e12[\s\S]+(largest-contentful-paint|first-input|layout-shift)|(largest-contentful-paint|first-input|layout-shift)[\s\S]+8999999999999[\s\S]+1e12)/m;
     return response_bodies.filter(har => {
         return webVitalsJSPattern.test(har.response_body);
     }).map(har => har.url);
 }
 
-return Promise.all([getLcpElement()]).then(lcp_elem_stats => {
+return Promise.all([getLcpElement()]).then(([lcp_elem_stats]) => {
+    const lcpUrl = lcp_elem_stats.url;
+    const rawDoc = getRawHtmlDocument();
+    let isLcpDiscoverable = null;
+    let isLcpPreloaded = null;
+    let responseObject = null;
+    let rawLcpElement = null;
+    const isLcpExternalResource = lcpUrl != '';
+    if (isLcpExternalResource) {
+        rawLcpElement = Array.from(rawDoc.querySelectorAll('picture source, img')).find(i => {
+            let src = i.src;
+            if (i.nodeName == 'SOURCE') {
+                src = new URL(i.srcset, location.href).href;
+            }
+
+            return src == lcpUrl;
+        });
```

Comment:
This confused me at first. But now 

```suggestion
    // Check if LCP resource reference is in the raw HTML (as opposed to being injected later by JS)
    if (isLcpExternalResource) {
        rawLcpElement = Array.from(rawDoc.querySelectorAll('picture source, img')).find(i => {
            let src = i.src;
            if (i.nodeName == 'SOURCE') {
                src = new URL(i.srcset, location.href).href;
            }

            return src == lcpUrl;
        });
```

Reply:
Added the comment inside the if statement